### DISCUSSION TITLE
How to set border-color on button click?

### DISCUSSION THREAD

**Discussion by gauravguptabits** [TomSchimansky/CustomTkinter#309] [General]
Hi,

I have a set of buttons in my UI, I would like to set the border-color of button when clicked and border-color of other button need to be unset as well.

Example

1. Add 3 buttons to the UI, (say start, stop, reset).
2. All buttons are initialized to have no border-color.
3. Click on 'start' button. This should set border-color of start button to black with some border-width (say 3).
4. Click on 'stop' button. This should unset the border-color of start button and set the border-color of stop button to black with some border-width (say 3).

My current implementation looks like following:
```
    def __init__():
        # some code above...
        self.start_stream_btn = customtkinter.CTkButton(master=self.frame,
                                                        fg_color='green',
                                                        text="Start Stream",
                                                        command=self.start_stream)
       # some code below...

    def _unset_btn_border(self):
        self.reset_btn.border_color = None
        self.quit_btn.border_color = None
        self.start_stream_btn.border_color = None
        self.stop_stream_btn.border_color = None

    def start_stream(self):
        self._unset_btn_border()
        self.start_stream_btn.border_color = 'black'
        self.start_stream_btn.border_width = 3

```

**Comment by guptakvgaurav** [comment]
Found the solution for it

```
self.btn.draw()
```
This function repaints the button with updated configuration

**Comment by TomSchimansky** [comment]
You shouldn't use the draw by yourself. I consider this a private method and I will indicate this by renaming it to _draw soo. But if you want to change the border_color or the border_width, you would use the configure method:
```python
self.start_stream_btn.configure(border_color=new_color, border_width=new_width)
```
